src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediatePortalStatement.java

```diff
@@ -37,21 +37,21 @@
   protected List<Short> parameterFormatCodes;
   protected List<Short> resultFormatCodes;
 
-  public IntermediatePortalStatement(
-      PreparedStatement statement,
-      String sql,
-      int parameterCount,
-      SetMultimap<Integer, Integer> parameterIndexToPositions,
-      Connection connection) {
-    super(statement, sql, parameterCount, parameterIndexToPositions, connection);
+  public IntermediatePortalStatement(String sql, Connection connection) {
+    super(sql, connection);
+    this.statement = Statement.of(sql);
     this.parameterFormatCodes = new ArrayList<>();
     this.resultFormatCodes = new ArrayList<>();
   }
 
+  void setBoundStatement(Statement statement) {
+    this.statement = statement;
+  }
+
   public short getParameterFormatCode(int index) {
     if (this.parameterFormatCodes.size() == 0) {
       return 0;
-    } else if (this.parameterFormatCodes.size() == 1) {
+    } else if (index >= this.parameterFormatCodes.size()) {
```

Comment:
The PG wire format spec says that if a parameter has a higher index than the list of format codes, the default should be used (the first format code given).